wsd/DocumentBroker.cpp

```diff
@@ -1397,6 +1397,13 @@ void DocumentBroker::handleSaveResponse(const std::shared_ptr<ClientSession>& se
         _nextStorageAttrs.setUserModified(wasModified);
     }
 
+    // Update the storage attributes to capture what's
+    // new and applies to this new version and reset the next.
+    // These are the attributes of the version to be uploaded.
+    // Note: these are owned by us and this is thread-safe.
+    _currentStorageAttrs.merge(_nextStorageAttrs);
```

Comment:
It was not -that- clear to me how these attributes are associated with an ongoing save operation; is it implied that there is only one?
I would assume that we have an existing asynchronous object to represent an ongoing save which this shadows - but on the DocumentBroker - is it in the right place ?

Reply:
Good questions. There are 3 sets of attributes now, to capture the attributes we are uploading (to allow for re-trying when we fail), to track the attributes of an in-flight save (to be merged with the uploading attributes once a new version is saved), and the "current" attributes for modifications that the user is making (while a save and an earlier upload might be happening in parallel).

The idea is to track the attributes for each of these 3 parallel activities separately, and only merge them when one activity fails and needs to retry with the new state of the doc/file.

DocumentBroker is the right place, because it's the only place where we track saving and uploading as well as the modification state of the document.